pkg/auth/dependency/sso/authurl.go

```diff
@@ -0,0 +1,37 @@
+package sso
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type authURLParams struct {
+	prividerName   string
+	clientID       string
+	urlPrefix      string
+	scope          Scope
+	options        Options
+	stateJWTSecret string
+	state          State
+	baseURL        string
+}
+
+func authURL(params authURLParams) (string, error) {
+	encodedState, err := EncodeState(params.stateJWTSecret, params.state)
+	if err != nil {
+		return "", err
+	}
+	v := url.Values{}
+	v.Set("response_type", "code")
+	v.Add("client_id", params.clientID)
+	v.Add("redirect_uri", RedirectURI(params.urlPrefix, params.prividerName))
+	v.Add("state", encodedState)
+	v.Add("scope", strings.Join(params.scope, " "))
+	v.Add("access_type", "offline")
+	v.Add("prompt", "select_account")
```

Comment:
I tried generate google auth url and found `access_type` and `prompt` params are repeated.

1. `access_type` and `prompt` should be for google only
1.  if any of the query params are provided in `params.options` we should use them, here should provide the default value only.

Reply:
@carmenlau `access_type` and `prompt` have added in `google.go` already, I accidentally added two lines of code again in `authurl.go`. Thanks for spotting this issue.